File: backend/src/main/java/com/matheus/book/application/book/dto/response/BookResponse.java
package com.matheus.book.application.book.dto.response;

import com.matheus.book.application.customer.dto.response.CustomerSimpleResponse;

import java.time.LocalDate;
import java.util.List;

public record BookResponse(
        Long id,
        String name,
        Integer quantity,
        Integer borrowedQuantity,
        String publisherName,
        LocalDate publishedIn,
        Long publisherId,
        List<CustomerSimpleResponse> customerSimpleResponses
) { }

File: frontend/src/app/home/api/book/dto/response/book-response.ts
import { CustomerSimpleResponse } from '../../../customer/dto/response/customer-simple-response';

export interface BookResponse {
  id: number;
  name: string;
  quantity: number;
  borrowedQuantity: number;
  publisherName: string;
  publishedIn: Date;
  publisherId: number;
  customerSimpleResponses: CustomerSimpleResponse[];
}
